### ISSUE TITLE
New field `index` to get child index 

### ISSUE THREAD

**Issue by nyancodeid** [nyancodeid/denoql#4]
I think we need a new working field to get index value from childNodes or Element.

Reference :
- https://stackoverflow.com/questions/5913927/get-child-node-index

**Comment by nyancodeid** [comment]
Okay, maybe this is a little tricky to realize. because when we use the code that is in the reference, we will not get the index value in order when there is another node in the middle of the parent childNodes.

```html
<div #parent>
  <p>Index 0</p>
  <p>Index 1</p>
  <div #other></div>
  <p>Index 2</p>
<div>
```
When we use `queryAll(selector: 'div#parent p')` we should have array of P Element. But, if we add index filed the last P Element actually have index `3` when we expected to be `2`.

**Comment by nyancodeid** [comment]
I think i will drop this new feature field.

**Comment by nyancodeid** [comment]
Solved with PR #8
